tests/query/parser/test_expressions.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    FunctionCall,
+    Literal,
+)
+from snuba.query.parser.expressions import parse_aggregation
+
+test_data = [
+    (
+        ["count", "event_id", None],
+        FunctionCall(None, "count", (Column(None, "event_id", None),)),
+    ),  # Simple aggregation
+    (
+        ["count()", "", None],
+        FunctionCall(None, "count", ()),
+    ),  # Common way to provide count()
+    (
+        ["count()", None, None],
+        FunctionCall(None, "count", ()),
+    ),  # Common way to provide count()
+    (
+        ["count()", "event_id", None],
+        CurriedFunctionCall(
+            None, FunctionCall(None, "count", ()), (Column(None, "event_id", None),)
+        ),
+    ),  # This is probably wrong, but we cannot disambiguate it at this level
```

Comment:
What would this be formatted like?

Reply:
> count()(eveent_id)

That's why I am saying this is wrong. 
it should either be 
> ["count", "event_id", None]
or
> ["count(event_id)", None, None]

Not sure this actually happens in prod.

Reply:
Ah okay — I assumed this was based off something actually happening. Thanks for clarifying.